test/unit/Lexer-spec.js

```diff
@@ -345,19 +345,27 @@ a | b
         md: `
 1. item 1
 2. item 2
+3) item 3
+4) item 4
```

Comment:
This unit test actually goes against the spec [example 272](https://spec.commonmark.org/0.29/#example-272). marked should start a new list if the bullet or ordered list delimiter is changed.

I'm ok with not fixing that spec example in this PR but can you create an integration test instead of changing the unit test. You can create a new `.md` file and `.html` file in `/test/specs/new/`

something like:

list_paren_delimiter.md
```md
1) one
2) two
3) three


2) two
3) three
4) four
```

list_paren_delimiter.html
```html
<ol>
<li>one</li>
<li>two</li>
<li>three</li>
</ol>

<ol start="2">
<li>two</li>
<li>three</li>
<li>four</li>
</ol>
```

Reply:
I've made changes to accommodate ex.272. I am not too sure what the `shouldFail` keyword means in the CM tests, but some of them had to be removed. In any case, the fixes work well.

Reply:
the `shouldFail` is there so we can track which spec tests are passing and which are failing.

Reply:
So `shouldFail` means `shouldNotFailButDoesFail`?

Reply:
exactly 😁👍